UM/Settings/MachineManager.py

```diff
@@ -150,7 +200,42 @@ def setActiveMachineInstance(self, machine):
 
         self._updateSettingVisibility(setting_visibility)
 
+        self._protect_working_profile = True
+        profile = self.findProfile(machine.getActiveProfileName(), instance = self._active_machine)
+        if profile:
+            self.setActiveProfile(profile)
+        else:
+            for profile in self._profiles:
+                self.setActiveProfile(profile) #default to first profile you can find
+                break
```

Comment:
Nothing.
I copied this approach from somewhere. I guess it could be rewritten as ```if len(self._profiles) > 0: self.setActiveProfile(self._profiles[0])```
Note that the case where ```self._profiles``` is empty should be very rare (if ever) because there will be either machine-specific or generic profiles. The generic profiles are basically always available.